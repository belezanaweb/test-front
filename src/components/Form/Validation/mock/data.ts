export const fields = [
  {
    id: 'account.name',
    label: 'Nome',
    validations: [
      {
        type: 'required'
      },
      {
        type: 'min',
        value: 3
      }
    ]
  },
  {
    id: 'email',
    label: 'E-mail',
    validations: [
      {
        type: 'required'
      },
      {
        type: 'min',
        value: 3
      },
      {
        type: 'email'
      }
    ]
  },
  {
    id: 'account.mother',
    label: 'Nome da mãe'
  },
  {
    id: 'account.cpf',
    label: 'CPF',
    validations: [
      {
        type: 'check_cpf'
      }
    ]
  },
  {
    id: 'account.phone.areaCode',
    label: 'Código de área',
    validations: [
      {
        type: 'required'
      },
      {
        type: 'only_numbers'
      }
    ]
  },
  {
    id: 'account.phone.number',
    label: 'Número do telefone',
    validations: [
      {
        type: 'required'
      },
      {
        type: 'only_numbers'
      }
    ]
  }
]

export const fieldWithoutId = {
  label: 'Endereço',
  validations: [
    {
      type: 'required'
    }
  ]
}

export const fieldWithoutLabel = {
  id: 'address',
  validations: [
    {
      type: 'required'
    }
  ]
}

export const values = {
  account: {
    name: '',
    phone: {
      areaCode: 'h',
      number: '969093321'
    },
    mother: 'Jumira',
    cpf: 'A34'
  },
  email: 'f'
}

export const validationReq = {
  type: 'required'
}

export const validationMin = {
  type: 'min',
  value: 3
}
