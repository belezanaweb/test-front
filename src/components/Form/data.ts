import { format } from 'date-fns'

import { IFields } from './interface'

export const fields: IFields[] = [
  {
    id: 'number',
    label: 'Número do cartão',
    placeholder: '___.____.____.____',
    type: 'text',
    mask: [
      /\d/,
      /\d/,
      /\d/,
      /\d/,
      '.',
      /\d/,
      /\d/,
      /\d/,
      /\d/,
      '.',
      /\d/,
      /\d/,
      /\d/,
      /\d/,
      '.',
      /\d/,
      /\d/,
      /\d/,
      /\d/
    ],
    validations: [
      {
        type: 'required'
      },
      {
        type: 'min',
        value: 16
      },
      {
        type: 'only_numbers'
      }
    ]
  },
  {
    id: 'holder',
    label: 'Nome do titular',
    placeholder: 'Como no cartão',
    type: 'text',
    validations: [
      {
        type: 'required'
      }
    ]
  },
  {
    id: 'expirationDate',
    label: 'Validade (mês/ano)',
    placeholder: '__/____',
    type: 'text',
    mask: [/\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
    validations: [
      {
        type: 'required'
      },
      {
        type: 'after_or_equal',
        value: format(new Date(), 'MM/yyyy')
      }
    ]
  },
  {
    id: 'cvv',
    label: 'CVV',
    placeholder: '__',
    type: 'text',
    mask: [/\d/, /\d/, /\d/],
    validations: [
      {
        type: 'required'
      },
      {
        type: 'size',
        value: 3
      },
      {
        type: 'only_numbers'
      }
    ]
  }
]
