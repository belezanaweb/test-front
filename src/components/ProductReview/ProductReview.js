import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

const StyledProduct = styled.div`
  display: flex;
  background-color: #fff;
  border-radius: 3px;
  border: 1px solid #eee;
  padding: 4px 6px 2px 10px;
  margin-bottom: 15px;
`

const StyledProductName = styled.p`
  color: #212122;
  font-size: 13px;
  line-height: 16px;
  word-spacing: 0.9px;
  margin-top: 10px;
  margin-bottom: 17px;
`

const StyledProductImg = styled.img`
  width: 44px;
  height: 100%;
  margin-right: 8px;
  margin-top: 7px;
`

function ProductReview({ product }) {
  return (
    <StyledProduct>
      <StyledProductImg src={product.imageObjects[0].thumbnail} alt={product.name} />
      <div>
        <StyledProductName>{product.name}</StyledProductName>
      </div>
    </StyledProduct>
  )
}

ProductReview.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    imageObjects: PropTypes.array
  }).isRequired
}

export default ProductReview
