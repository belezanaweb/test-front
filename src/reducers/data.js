const INITIAL_STATE = {
  isWorking: 'Ok'
}

const data = (state = INITIAL_STATE, { type, payload }) => {
  return state
}

export default data
