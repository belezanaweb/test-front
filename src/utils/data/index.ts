export const maskCardNumber = (cardNumber: string) => {
  const value = cardNumber.replace(/\s/g, '')

  if (value.length === 16) {
    return value.slice(-4).padStart(16, '*')
  }

  return ''
}
