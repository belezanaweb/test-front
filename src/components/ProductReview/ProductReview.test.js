import React from 'react'
import { render } from '@testing-library/react'

import data from '../../data/endpoint.json'

import ProductReview from './ProductReview'

describe('ProductReview Component', () => {
  test('renders links', async () => {
    const { asFragment } = render(<ProductReview product={data.items[0].product} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
