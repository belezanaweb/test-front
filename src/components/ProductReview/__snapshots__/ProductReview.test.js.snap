// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReview Component renders links 1`] = `
<DocumentFragment>
  <div
    class="ProductReview__StyledProduct-sc-2kzi8k-0 jOZocn"
  >
    <img
      alt="L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium"
      class="ProductReview__StyledProductImg-sc-2kzi8k-2 SwLGV"
      src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png"
    />
    <div>
      <p
        class="ProductReview__StyledProductName-sc-2kzi8k-1 jeElbh"
      >
        L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium
      </p>
    </div>
  </div>
</DocumentFragment>
`;
